api/swagger.yml

```
@@ -2589,7 +2610,54 @@ paths:
           description: too many requests
         default:
           $ref: "#/components/responses/ServerError"
-
+    post:
+      tags:
+        - internal
+      operationId: setRepositoryMetadata
+      summary: set repository metadata
+      description: |
+        Set repository metadata. This will only add or update the provided keys, and will not remove any existing keys.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RepositoryMetadataSet"
+      responses:
+        204:
+          description: repository metadata set successfully
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+    delete:
+      tags:
+        - internal
+      operationId: deleteRepositoryMetadata
+      summary: delete repository metadata
+      description: |
+        Delete repository metadata. Deletes the provided keys from the repository metadata.
```

```suggestion
        Delete specified keys from the repository's metadata.
```

Also describe if we use 404 in case key to delete not found (I assume not)

---

We don't, removing